# How to Meet and Retrieve Your Data

Watershed's intern put together a couple of sources of information that will be useful for your project. These three types of information are contained in the capstone database:
1. the current monthly rent Watershed charges for all of their client’s 244 properties, as well as the property type and geographic location of those properties. 
2. some general information about examples of short-term rental properties.  This information can be used to get a sense of what kind of nightly rental price Watershed’s client’s properties *could* be listed for, if they were converted to short-term rentals.
3. records about when those short-term rental properties were rented out, so that you can calculate their occupancy rates.  

Your job is to determine how the database is organized so that you can retrieve all of the available information about Watershed’s client’s 244 properties, as well as the corresponding short-term rental information for comparable properties in the same location and of the same type.
1. Start by determining what tables the database contains, and what fields are included in each table.  
2. Then, we recommend that you make at least a rough relational schema of how the database is organized, so that you know what fields you can use to join tables. 
3. Next, make a list of the columns of data you want to retrieve in your final output.  
4. Finally, write your query to retrieve the desired data from the database.  

### Load and connect to the database

In [1]:
%load_ext sql
%sql mysql://studentuser:studentpw@mysqlserver/capstone 
%sql USE capstone

0 rows affected.


[]

# Relational schema

# Explore the dataset

In [2]:
%%sql
SHOW tables

6 rows affected.


[('location',),
 ('property_type',),
 ('st_property_info',),
 ('st_rental_dates',),
 ('st_rental_prices',),
 ('watershed_property_info',)]

In [3]:
%%sql
DESCRIBE location

4 rows affected.


[('location_id', 'varchar(255)', 'NO', 'PRI', None, ''),
 ('city', 'varchar(255)', 'YES', '', None, ''),
 ('state', 'varchar(255)', 'YES', '', None, ''),
 ('zipcode', 'int(5)', 'YES', '', None, '')]

In [4]:
%%sql
SELECT  *
FROM location
LIMIT 10

10 rows affected.


[('L1', 'Anchorage', 'AK', 99501),
 ('L10', 'Anchorage', 'AK', 99518),
 ('L100', 'Cherokee', 'AL', 35616),
 ('L1000', 'Azusa', 'CA', 91702),
 ('L10000', 'Winston-Sa', 'NC', 27101),
 ('L10001', 'Winston-Sa', 'NC', 27103),
 ('L10002', 'Winston-Sa', 'NC', 27104),
 ('L10003', 'Winston-Sa', 'NC', 27105),
 ('L10004', 'Winston-Sa', 'NC', 27106),
 ('L10005', 'Winston-Sa', 'NC', 27107)]

In [5]:
%%sql
DESCRIBE property_type

5 rows affected.


[('property_type_id', 'varchar(255)', 'NO', 'PRI', None, ''),
 ('apt_house', 'varchar(255)', 'YES', '', None, ''),
 ('num_bedrooms', 'varchar(255)', 'YES', '', None, ''),
 ('kitchen', 'varchar(255)', 'YES', '', None, ''),
 ('shared', 'varchar(255)', 'YES', '', None, '')]

In [6]:
%%sql
SELECT  *
FROM property_type
LIMIT 10

10 rows affected.


[('R1', 'apartment', '1', 'Y', 'Y'),
 ('R10', 'house', '1', 'Y', 'N'),
 ('R11', 'house', '1', 'N', 'Y'),
 ('R12', 'house', '1', 'N', 'N'),
 ('R13', 'house', '2', 'Y', 'Y'),
 ('R14', 'house', '2', 'Y', 'N'),
 ('R15', 'house', '2', 'N', 'Y'),
 ('R16', 'house', '2', 'N', 'N'),
 ('R2', 'apartment', '1', 'Y', 'N'),
 ('R3', 'apartment', '1', 'N', 'Y')]

In [7]:
%sql DESCRIBE st_property_info

3 rows affected.


[('st_property_id', 'varchar(255)', 'NO', 'PRI', None, ''),
 ('location', 'varchar(255)', 'YES', '', None, ''),
 ('property_type', 'varchar(255)', 'YES', '', None, '')]

In [8]:
%%sql
SELECT  *
FROM st_property_info
LIMIT 10

10 rows affected.


[('ST1', 'L9531', 'R6'),
 ('ST10', 'L9533', 'R6'),
 ('ST100', 'L1944', 'R2'),
 ('ST1000', 'L5957', 'R1'),
 ('ST101', 'L15257', 'R2'),
 ('ST102', 'L15257', 'R6'),
 ('ST103', 'L15257', 'R10'),
 ('ST104', 'L15257', 'R14'),
 ('ST105', 'L15260', 'R2'),
 ('ST106', 'L15260', 'R6')]

In [9]:
%sql DESCRIBE st_rental_dates

2 rows affected.


[('rental_date', 'date', 'NO', 'PRI', None, ''),
 ('st_property', 'varchar(255)', 'NO', 'PRI', None, '')]

In [10]:
%%sql
SELECT  *
FROM st_rental_dates
LIMIT 10

10 rows affected.


[(datetime.date(2014, 1, 31), 'ST100'),
 (datetime.date(2014, 1, 31), 'ST106'),
 (datetime.date(2014, 1, 31), 'ST107'),
 (datetime.date(2014, 1, 31), 'ST109'),
 (datetime.date(2014, 1, 31), 'ST113'),
 (datetime.date(2014, 1, 31), 'ST114'),
 (datetime.date(2014, 1, 31), 'ST116'),
 (datetime.date(2014, 1, 31), 'ST118'),
 (datetime.date(2014, 1, 31), 'ST119'),
 (datetime.date(2014, 1, 31), 'ST123')]

In [11]:
%sql DESCRIBE st_rental_prices

5 rows affected.


[('location', 'varchar(255)', 'NO', 'PRI', None, ''),
 ('property_type', 'varchar(255)', 'NO', 'PRI', None, ''),
 ('percentile_10th_price', 'int(11)', 'YES', '', None, ''),
 ('percentile_90th_price', 'int(11)', 'YES', '', None, ''),
 ('sample_nightly_rent_price', 'int(11)', 'YES', '', None, '')]

In [12]:
%%sql
SELECT  *
FROM st_rental_prices
LIMIT 10

10 rows affected.


[('L10126', 'R1', 83, 273, 192),
 ('L10126', 'R10', 155, 494, 257),
 ('L10126', 'R11', 82, 411, 242),
 ('L10126', 'R12', 119, 366, 229),
 ('L10126', 'R13', 137, 416, 375),
 ('L10126', 'R14', 151, 391, 286),
 ('L10126', 'R15', 116, 312, 172),
 ('L10126', 'R16', 126, 343, 258),
 ('L10126', 'R2', 91, 342, 229),
 ('L10126', 'R3', 41, 242, 117)]

In [13]:
%sql DESCRIBE watershed_property_info

4 rows affected.


[('ws_property_id', 'varchar(255)', 'NO', 'PRI', None, ''),
 ('location', 'varchar(255)', 'YES', '', None, ''),
 ('property_type', 'varchar(255)', 'YES', '', None, ''),
 ('current_monthly_rent', 'int(11)', 'NO', '', None, '')]

In [14]:
%%sql
SELECT  *
FROM watershed_property_info
LIMIT 10

10 rows affected.


[('W1', 'L9531', 'R6', 1060),
 ('W10', 'L9533', 'R6', 1200),
 ('W100', 'L1944', 'R2', 3300),
 ('W101', 'L15257', 'R2', 1400),
 ('W102', 'L15257', 'R6', 2000),
 ('W103', 'L15257', 'R10', 1600),
 ('W104', 'L15257', 'R14', 2800),
 ('W105', 'L15260', 'R2', 1100),
 ('W106', 'L15260', 'R6', 1900),
 ('W107', 'L15260', 'R10', 1800)]

In [15]:
%%sql
SELECT *
FROM watershed_property_info ws
LEFT JOIN property_type pt ON ws.property_type = pt.property_type_id
LEFT JOIN location loc ON ws.location = loc.location_id

244 rows affected.


[('W1', 'L9531', 'R6', 1060, 'R6', 'apartment', '2', 'Y', 'N', 'L9531', 'Chapel Hill', 'NC', 27514),
 ('W10', 'L9533', 'R6', 1200, 'R6', 'apartment', '2', 'Y', 'N', 'L9533', 'Chapel Hill', 'NC', 27517),
 ('W100', 'L1944', 'R2', 3300, 'R2', 'apartment', '1', 'Y', 'N', 'L1944', 'San Francisco', 'CA', 94129),
 ('W101', 'L15257', 'R2', 1400, 'R2', 'apartment', '1', 'Y', 'N', 'L15257', 'Austin', 'TX', 78702),
 ('W102', 'L15257', 'R6', 2000, 'R6', 'apartment', '2', 'Y', 'N', 'L15257', 'Austin', 'TX', 78702),
 ('W103', 'L15257', 'R10', 1600, 'R10', 'house', '1', 'Y', 'N', 'L15257', 'Austin', 'TX', 78702),
 ('W104', 'L15257', 'R14', 2800, 'R14', 'house', '2', 'Y', 'N', 'L15257', 'Austin', 'TX', 78702),
 ('W105', 'L15260', 'R2', 1100, 'R2', 'apartment', '1', 'Y', 'N', 'L15260', 'Austin', 'TX', 78705),
 ('W106', 'L15260', 'R6', 1900, 'R6', 'apartment', '2', 'Y', 'N', 'L15260', 'Austin', 'TX', 78705),
 ('W107', 'L15260', 'R10', 1800, 'R10', 'house', '1', 'Y', 'N', 'L15260', 'Austin', 'TX', 78705),
 ('W108', 'L15260', 'R14', 3200, 'R14', 'house', '2', 'Y', 'N', 'L15260', 'Austin', 'TX', 78705),
 ('W109', 'L15264', 'R2', 1000, 'R2', 'apartment', '1', 'Y', 'N', 'L15264', 'Austin', 'TX', 78723),
 ('W11', 'L9533', 'R10', 1000, 'R10', 'house', '1', 'Y', 'N', 'L9533', 'Chapel Hill', 'NC', 27517),
 ('W110', 'L15264', 'R6', 1300, 'R6', 'apartment', '2', 'Y', 'N', 'L15264', 'Austin', 'TX', 78723),
 ('W111', 'L15264', 'R10', 1200, 'R10', 'house', '1', 'Y', 'N', 'L15264', 'Austin', 'TX', 78723),
 ('W112', 'L15264', 'R14', 1600, 'R14', 'house', '2', 'Y', 'N', 'L15264', 'Austin', 'TX', 78723),
 ('W113', 'L15278', 'R2', 800, 'R2', 'apartment', '1', 'Y', 'N', 'L15278', 'Austin', 'TX', 78744),
 ('W114', 'L15278', 'R6', 1200, 'R6', 'apartment', '2', 'Y', 'N', 'L15278', 'Austin', 'TX', 78744),
 ('W115', 'L15278', 'R10', 900, 'R10', 'house', '1', 'Y', 'N', 'L15278', 'Austin', 'TX', 78744),
 ('W116', 'L15278', 'R14', 1100, 'R14', 'house', '2', 'Y', 'N', 'L15278', 'Austin', 'TX', 78744),
 ('W117', 'L15280', 'R2', 1000, 'R2', 'apartment', '1', 'Y', 'N', 'L15280', 'Austin', 'TX', 78746),
 ('W118', 'L15280', 'R6', 1400, 'R6', 'apartment', '2', 'Y', 'N', 'L15280', 'Austin', 'TX', 78746),
 ('W119', 'L15280', 'R10', 1500, 'R10', 'house', '1', 'Y', 'N', 'L15280', 'Austin', 'TX', 78746),
 ('W12', 'L9533', 'R14', 1300, 'R14', 'house', '2', 'Y', 'N', 'L9533', 'Chapel Hill', 'NC', 27517),
 ('W120', 'L15280', 'R14', 1600, 'R14', 'house', '2', 'Y', 'N', 'L15280', 'Austin', 'TX', 78746),
 ('W121', 'L463', 'R2', 600, 'R2', 'apartment', '1', 'Y', 'N', 'L463', 'Bentonville', 'AR', 72712),
 ('W122', 'L463', 'R6', 800, 'R6', 'apartment', '2', 'Y', 'N', 'L463', 'Bentonville', 'AR', 72712),
 ('W123', 'L463', 'R10', 700, 'R10', 'house', '1', 'Y', 'N', 'L463', 'Bentonville', 'AR', 72712),
 ('W124', 'L463', 'R14', 1000, 'R14', 'house', '2', 'Y', 'N', 'L463', 'Bentonville', 'AR', 72712),
 ('W125', 'L464', 'R2', 700, 'R2', 'apartment', '1', 'Y', 'N', 'L464', 'Bentonville', 'AR', 72719),
 ('W126', 'L464', 'R6', 900, 'R6', 'apartment', '2', 'Y', 'N', 'L464', 'Bentonville', 'AR', 72719),
 ('W127', 'L464', 'R10', 1000, 'R10', 'house', '1', 'Y', 'N', 'L464', 'Bentonville', 'AR', 72719),
 ('W128', 'L464', 'R14', 1200, 'R14', 'house', '2', 'Y', 'N', 'L464', 'Bentonville', 'AR', 72719),
 ('W129', 'L2314', 'R2', 1200, 'R2', 'apartment', '1', 'Y', 'N', 'L2314', 'Denver', 'CO', 80204),
 ('W13', 'L9534', 'R6', 920, 'R6', 'apartment', '2', 'Y', 'N', 'L9534', 'Chapel Hill', 'NC', 27707),
 ('W130', 'L2314', 'R6', 1300, 'R6', 'apartment', '2', 'Y', 'N', 'L2314', 'Denver', 'CO', 80204),
 ('W131', 'L2314', 'R10', 1100, 'R10', 'house', '1', 'Y', 'N', 'L2314', 'Denver', 'CO', 80204),
 ('W132', 'L2314', 'R14', 1200, 'R14', 'house', '2', 'Y', 'N', 'L2314', 'Denver', 'CO', 80204),
 ('W133', 'L2318', 'R2', 1300, 'R2', 'apartment', '1', 'Y', 'N', 'L2318', 'Denver', 'CO', 80209),
 ('W134', 'L2318', 'R6', 1700, 'R6', 'apartment', '2', 'Y', 'N', 'L2318', 'Denver', 'CO', 80209),
 ('W135', 'L2318', 'R10', 1200, 'R

In [16]:
%%sql
SELECT ws.ws_property_id AS ws_id, ws.location AS ws_location, ws.ws_property_id AS ws_id, ws.property_type AS ws_type, 
        strp.sample_nightly_rent_price AS sample_price
FROM watershed_property_info ws
LEFT JOIN st_rental_prices strp ON strp.location = ws.location AND strp.property_type = ws.property_type

244 rows affected.


[('W1', 'L9531', 'W1', 'R6', 148),
 ('W10', 'L9533', 'W10', 'R6', 133),
 ('W100', 'L1944', 'W100', 'R2', 372),
 ('W101', 'L15257', 'W101', 'R2', 302),
 ('W102', 'L15257', 'W102', 'R6', 429),
 ('W103', 'L15257', 'W103', 'R10', 380),
 ('W104', 'L15257', 'W104', 'R14', 374),
 ('W105', 'L15260', 'W105', 'R2', 386),
 ('W106', 'L15260', 'W106', 'R6', 212),
 ('W107', 'L15260', 'W107', 'R10', 969),
 ('W108', 'L15260', 'W108', 'R14', 885),
 ('W109', 'L15264', 'W109', 'R2', 287),
 ('W11', 'L9533', 'W11', 'R10', 206),
 ('W110', 'L15264', 'W110', 'R6', 462),
 ('W111', 'L15264', 'W111', 'R10', 389),
 ('W112', 'L15264', 'W112', 'R14', 678),
 ('W113', 'L15278', 'W113', 'R2', 163),
 ('W114', 'L15278', 'W114', 'R6', 374),
 ('W115', 'L15278', 'W115', 'R10', 444),
 ('W116', 'L15278', 'W116', 'R14', 426),
 ('W117', 'L15280', 'W117', 'R2', 332),
 ('W118', 'L15280', 'W118', 'R6', 430),
 ('W119', 'L15280', 'W119', 'R10', 662),
 ('W12', 'L9533', 'W12', 'R14', 186),
 ('W120', 'L15280', 'W120', 'R14', 696),
 ('W121', 'L463', 'W121', 'R2', 182),
 ('W122', 'L463', 'W122', 'R6', 241),
 ('W123', 'L463', 'W123', 'R10', 363),
 ('W124', 'L463', 'W124', 'R14', 301),
 ('W125', 'L464', 'W125', 'R2', 212),
 ('W126', 'L464', 'W126', 'R6', 340),
 ('W127', 'L464', 'W127', 'R10', 266),
 ('W128', 'L464', 'W128', 'R14', 442),
 ('W129', 'L2314', 'W129', 'R2', 354),
 ('W13', 'L9534', 'W13', 'R6', 123),
 ('W130', 'L2314', 'W130', 'R6', 377),
 ('W131', 'L2314', 'W131', 'R10', 318),
 ('W132', 'L2314', 'W132', 'R14', 198),
 ('W133', 'L2318', 'W133', 'R2', 149),
 ('W134', 'L2318', 'W134', 'R6', 210),
 ('W135', 'L2318', 'W135', 'R10', 187),
 ('W136', 'L2318', 'W136', 'R14', 225),
 ('W137', 'L2323', 'W137', 'R2', 123),
 ('W138', 'L2323', 'W138', 'R6', 263),
 ('W139', 'L2323', 'W139', 'R10', 238),
 ('W14', 'L9534', 'W14', 'R10', 146),
 ('W140', 'L2323', 'W140', 'R14', 349),
 ('W141', 'L2325', 'W141', 'R2', 147),
 ('W142', 'L2325', 'W142', 'R6', 151),
 ('W143', 'L2325', 'W143', 'R10', 429),
 ('W144', 'L2325', 'W144', 'R14', 441),
 ('W145', 'L2338', 'W145', 'R2', 144),
 ('W146', 'L2338', 'W146', 'R6', 136),
 ('W147', 'L2338', 'W147', 'R10', 305),
 ('W148', 'L2338', 'W148', 'R14', 425),
 ('W149', 'L3244', 'W149', 'R2', 176),
 ('W15', 'L9534', 'W15', 'R14', 169),
 ('W150', 'L3244', 'W150', 'R6', 207),
 ('W151', 'L3244', 'W151', 'R10', 244),
 ('W152', 'L3244', 'W152', 'R14', 536),
 ('W153', 'L3256', 'W153', 'R2', 476),
 ('W154', 'L3256', 'W154', 'R6', 360),
 ('W155', 'L3256', 'W155', 'R10', 1477),
 ('W156', 'L3256', 'W156', 'R14', 1265),
 ('W157', 'L3261', 'W157', 'R2', 328),
 ('W158', 'L3261', 'W158', 'R6', 246),
 ('W159', 'L3261', 'W159', 'R10', 325),
 ('W16', 'L9534', 'W16', 'R2', 94),
 ('W160', 'L3261', 'W160', 'R14', 428),
 ('W161', 'L3262', 'W161', 'R2', 188),
 ('W162', 'L3262', 'W162', 'R6', 274),
 ('W163', 'L3262', 'W163', 'R10', 860),
 ('W164', 'L3262', 'W164', 'R14', 729),
 ('W165', 'L3264', 'W165', 'R2', 174),
 ('W166', 'L3264', 'W166', 'R6', 308),
 ('W167', 'L3264', 'W167', 'R10', 308),
 ('W168', 'L3264', 'W168', 'R14', 342),
 ('W169', 'L10126', 'W169', 'R2', 229),
 ('W17', 'L4761', 'W17', 'R6', 392),
 ('W170', 'L10126', 'W170', 'R6', 322),
 ('W171', 'L10126', 'W171', 'R10', 257),
 ('W172', 'L10126', 'W172', 'R14', 286),
 ('W173', 'L10130', 'W173', 'R2', 180),
 ('W174', 'L10130', 'W174', 'R6', 230),
 ('W175', 'L10130', 'W175', 'R10', 221),
 ('W176', 'L10130', 'W176', 'R14', 316),
 ('W177', 'L10133', 'W177', 'R2', 245),
 ('W178', 'L10133', 'W178', 'R6', 266),
 ('W179', 'L10133', 'W179', 'R10', 325),
 ('W18', 'L4761', 'W18', 'R10', 393),
 ('W180', 'L10133', 'W180', 'R14', 256),
 ('W181', 'L10136', 'W181', 'R2', 184),
 ('W182', 'L10136', 'W182', 'R6', 427),
 ('W183', 'L10136', 'W183', 'R10', 418),
 ('W184', 'L10136', 'W184', 'R14', 219),
 ('W185', 'L1882', 'W185', 'R2', 220),
 ('W186', 'L1882', 'W186', 'R6', 481),
 ('W187', 'L1882', 'W187', 'R10', 280),
 ('W188', 'L1882', 'W188', 'R14', 568),
 ('W189', 'L1883', 'W189', 'R2', 318),
 ('W19', 'L4761', 'W19', 'R14', 556),
 ('W19

In [17]:
%%sql
SELECT ws.ws_property_id AS ws_id, ws.location AS ws_location, ws.ws_property_id AS ws_id, ws.property_type AS ws_type, 
        sti.st_property_id AS samplt_id, strp.sample_nightly_rent_price AS sample_price, COUNT(DISTINCT strd.rental_date)/365 AS occupancy_rates 
FROM watershed_property_info ws
LEFT JOIN st_rental_prices strp ON strp.location = ws.location AND strp.property_type = ws.property_type
LEFT JOIN st_property_info sti ON sti.location = ws.location AND sti.property_type = ws.property_type
LEFT JOIN st_rental_dates strd ON strd.st_property = sti.st_property_id
WHERE ws.ws_property_id = 'W24' AND YEAR(strd.rental_date) = 2015
GROUP BY sti.st_property_id

1 rows affected.


[('W24', 'L4765', 'W24', 'R2', 'ST424', 239, Decimal('0.6767'))]

## Combine all datasets

In [18]:
%%sql
SELECT ws_property_id, properties.location_id, properties.property_type_id, current_monthly_rent, apt_house, num_bedrooms, kitchen, shared, city, state, zipcode,
       st_property_id, percentile_10th_price, percentile_90th_price, sample_nightly_rent_price, occupancy_rate
FROM
(SELECT *
 FROM watershed_property_info ws
 LEFT JOIN property_type pt ON ws.property_type = pt.property_type_id
 LEFT JOIN location loc ON ws.location = loc.location_id) AS properties
LEFT JOIN st_rental_prices strp ON strp.location = properties.location AND strp.property_type = properties.property_type
LEFT JOIN st_property_info sti ON sti.location = properties.location AND sti.property_type = properties.property_type
LEFT JOIN 
(SELECT st_property, COUNT(DISTINCT rental_date)/365 AS occupancy_rate
 FROM st_rental_dates
 WHERE YEAR(rental_date) = 2015
 GROUP BY st_property) AS occupancy
ON occupancy.st_property = sti.st_property_id

244 rows affected.


[('W1', 'L9531', 'R6', 1060, 'apartment', '2', 'Y', 'N', 'Chapel Hill', 'NC', 27514, 'ST1', 114, 153, 148, Decimal('0.1616')),
 ('W10', 'L9533', 'R6', 1200, 'apartment', '2', 'Y', 'N', 'Chapel Hill', 'NC', 27517, 'ST10', 111, 149, 133, Decimal('0.3479')),
 ('W100', 'L1944', 'R2', 3300, 'apartment', '1', 'Y', 'N', 'San Francisco', 'CA', 94129, 'ST100', 108, 610, 372, Decimal('0.3973')),
 ('W101', 'L15257', 'R2', 1400, 'apartment', '1', 'Y', 'N', 'Austin', 'TX', 78702, 'ST101', 178, 533, 302, Decimal('0.3644')),
 ('W102', 'L15257', 'R6', 2000, 'apartment', '2', 'Y', 'N', 'Austin', 'TX', 78702, 'ST102', 221, 617, 429, Decimal('0.4110')),
 ('W103', 'L15257', 'R10', 1600, 'house', '1', 'Y', 'N', 'Austin', 'TX', 78702, 'ST103', 202, 646, 380, Decimal('0.4110')),
 ('W104', 'L15257', 'R14', 2800, 'house', '2', 'Y', 'N', 'Austin', 'TX', 78702, 'ST104', 197, 639, 374, Decimal('0.5260')),
 ('W105', 'L15260', 'R2', 1100, 'apartment', '1', 'Y', 'N', 'Austin', 'TX', 78705, 'ST105', 114, 477, 386, Decimal('0.4329')),
 ('W106', 'L15260', 'R6', 1900, 'apartment', '2', 'Y', 'N', 'Austin', 'TX', 78705, 'ST106', 80, 583, 212, Decimal('0.6959')),
 ('W107', 'L15260', 'R10', 1800, 'house', '1', 'Y', 'N', 'Austin', 'TX', 78705, 'ST107', 239, 1431, 969, Decimal('0.1096')),
 ('W108', 'L15260', 'R14', 3200, 'house', '2', 'Y', 'N', 'Austin', 'TX', 78705, 'ST108', 236, 1533, 885, Decimal('0.2247')),
 ('W109', 'L15264', 'R2', 1000, 'apartment', '1', 'Y', 'N', 'Austin', 'TX', 78723, 'ST109', 138, 550, 287, Decimal('0.2192')),
 ('W11', 'L9533', 'R10', 1000, 'house', '1', 'Y', 'N', 'Chapel Hill', 'NC', 27517, 'ST11', 116, 296, 206, Decimal('0.3918')),
 ('W110', 'L15264', 'R6', 1300, 'apartment', '2', 'Y', 'N', 'Austin', 'TX', 78723, 'ST110', 175, 917, 462, Decimal('0.5370')),
 ('W111', 'L15264', 'R10', 1200, 'house', '1', 'Y', 'N', 'Austin', 'TX', 78723, 'ST111', 130, 821, 389, Decimal('0.5123')),
 ('W112', 'L15264', 'R14', 1600, 'house', '2', 'Y', 'N', 'Austin', 'TX', 78723, 'ST112', 241, 866, 678, Decimal('0.3616')),
 ('W113', 'L15278', 'R2', 800, 'apartment', '1', 'Y', 'N', 'Austin', 'TX', 78744, 'ST113', 134, 288, 163, Decimal('0.8438')),
 ('W114', 'L15278', 'R6', 1200, 'apartment', '2', 'Y', 'N', 'Austin', 'TX', 78744, 'ST114', 234, 794, 374, Decimal('0.9151')),
 ('W115', 'L15278', 'R10', 900, 'house', '1', 'Y', 'N', 'Austin', 'TX', 78744, 'ST115', 252, 547, 444, Decimal('0.4301')),
 ('W116', 'L15278', 'R14', 1100, 'house', '2', 'Y', 'N', 'Austin', 'TX', 78744, 'ST116', 246, 616, 426, Decimal('0.4822')),
 ('W117', 'L15280', 'R2', 1000, 'apartment', '1', 'Y', 'N', 'Austin', 'TX', 78746, 'ST117', 171, 457, 332, Decimal('0.4904')),
 ('W118', 'L15280', 'R6', 1400, 'apartment', '2', 'Y', 'N', 'Austin', 'TX', 78746, 'ST118', 262, 567, 430, Decimal('0.5233')),
 ('W119', 'L15280', 'R10', 1500, 'house', '1', 'Y', 'N', 'Austin', 'TX', 78746, 'ST119', 229, 859, 662, Decimal('0.4493')),
 ('W12', 'L9533', 'R14', 1300, 'house', '2', 'Y', 'N', 'Chapel Hill', 'NC', 27517, 'ST12', 136, 336, 186, Decimal('0.6603')),
 ('W120', 'L15280', 'R14', 1600, 'house', '2', 'Y', 'N', 'Austin', 'TX', 78746, 'ST120', 449, 899, 696, Decimal('0.4877')),
 ('W121', 'L463', 'R2', 600, 'apartment', '1', 'Y', 'N', 'Bentonville', 'AR', 72712, 'ST121', 132, 226, 182, Decimal('0.4384')),
 ('W122', 'L463', 'R6', 800, 'apartment', '2', 'Y', 'N', 'Bentonville', 'AR', 72712, 'ST122', 157, 340, 241, Decimal('0.5315')),
 ('W123', 'L463', 'R10', 700, 'house', '1', 'Y', 'N', 'Bentonville', 'AR', 72712, 'ST123', 215, 377, 363, Decimal('0.1397')),
 ('W124', 'L463', 'R14', 1000, 'house', '2', 'Y', 'N', 'Bentonville', 'AR', 72712, 'ST124', 202, 374, 301, Decimal('0.4685')),
 ('W125', 'L464', 'R2', 700, 'apartment', '1', 'Y', 'N', 'Bentonville', 'AR', 72719, 'ST125', 94, 356, 212, Decimal('0.5014')),
 ('W126', 'L464', 'R6', 900, 'apartment', '2', 'Y', 'N', 'Bentonville', 'AR', 72719, 'ST126', 69, 485, 340, Decimal('0.3068')),
 ('W127', 'L464', 'R10', 1000, 'house', '1', 'Y', 'N', 'Bentonville', 'AR', 72719, 

In [19]:
dataset = %%sql \
SELECT ws_property_id, properties.location_id, properties.property_type_id, current_monthly_rent, apt_house, num_bedrooms, kitchen, shared, city, state, zipcode, \
       st_property_id, percentile_10th_price, percentile_90th_price, sample_nightly_rent_price, occupancy_rate \
FROM \
(SELECT * \
 FROM watershed_property_info ws \
 LEFT JOIN property_type pt ON ws.property_type = pt.property_type_id \
 LEFT JOIN location loc ON ws.location = loc.location_id) AS properties \
LEFT JOIN st_rental_prices strp ON strp.location = properties.location AND strp.property_type = properties.property_type \
LEFT JOIN st_property_info sti ON sti.location = properties.location AND sti.property_type = properties.property_type \
LEFT JOIN  \
(SELECT st_property, COUNT(DISTINCT rental_date)/365 AS occupancy_rate \
 FROM st_rental_dates \
 WHERE YEAR(rental_date) = 2015 \
 GROUP BY st_property) AS occupancy \
ON occupancy.st_property = sti.st_property_id \

244 rows affected.


In [20]:
dataset.csv('project_dataset.csv')

CSV results at /home/jovyan/work/Capstone_course_materials/project_dataset.csv

# Quizes

**Q1: How many tables are there in the Watershed database?**

In [21]:
%%sql
SHOW tables

6 rows affected.


[('location',),
 ('property_type',),
 ('st_property_info',),
 ('st_rental_dates',),
 ('st_rental_prices',),
 ('watershed_property_info',)]

**Q2: How many fields, or columns, are there in the property_type table?**

In [22]:
%%sql
DESCRIBE property_type

5 rows affected.


[('property_type_id', 'varchar(255)', 'NO', 'PRI', None, ''),
 ('apt_house', 'varchar(255)', 'YES', '', None, ''),
 ('num_bedrooms', 'varchar(255)', 'YES', '', None, ''),
 ('kitchen', 'varchar(255)', 'YES', '', None, ''),
 ('shared', 'varchar(255)', 'YES', '', None, '')]

**Q3: Which of the following things are true about Watershed property W106?**  

A. It’s a 1-bedroom house in San Francisco, CA  

B. It’s a 2-bedroom apartment in Austin, TX  

C. It’s a 1-bedroom apartment in Miami, FL  

D. It’s a 2-bedroom house in Chapel Hill, NC

In [23]:
%%sql
SELECT *
FROM watershed_property_info ws
LEFT JOIN property_type pt ON ws.property_type = pt.property_type_id
LEFT JOIN location loc ON ws.location = loc.location_id
WHERE ws_property_id = 'W106'

1 rows affected.


[('W106', 'L15260', 'R6', 1900, 'R6', 'apartment', '2', 'Y', 'N', 'L15260', 'Austin', 'TX', 78705)]

**Q4: Which of the following things are true about the example short-term rental property that has the same type and is found in the same location as Watershed Property W24?**  

A. The nightly rental price is $180 and it’s occupancy rate in 2015 was 71.5%

B. The nightly rental price is $220 and it’s occupancy rate in 2015 was 40.3%

C. The nightly rental price is $239 and it’s occupancy rate in 2015 was 67.7%

D. The nightly rental price is $283 and it’s occupancy rate in 2015 was 49.9%  

In [24]:
%%sql
SELECT ws.ws_property_id AS ws_id, ws.location AS ws_location, ws.ws_property_id AS ws_id, ws.property_type AS ws_type, 
        sti.st_property_id AS samplt_id, strp.sample_nightly_rent_price AS sample_price, COUNT(DISTINCT strd.rental_date)/365 AS occupancy_rates 
FROM watershed_property_info ws
LEFT JOIN st_rental_prices strp ON strp.location = ws.location AND strp.property_type = ws.property_type
LEFT JOIN st_property_info sti ON sti.location = ws.location AND sti.property_type = ws.property_type
LEFT JOIN st_rental_dates strd ON strd.st_property = sti.st_property_id
WHERE ws.ws_property_id = 'W24' AND YEAR(strd.rental_date) = 2015
GROUP BY sti.st_property_id

1 rows affected.


[('W24', 'L4765', 'W24', 'R2', 'ST424', 239, Decimal('0.6767'))]

** Q5: How many of the client’s 244 Watershed properties have kitchens?**

In [25]:
%%sql
SELECT COUNT(*)
FROM watershed_property_info ws
LEFT JOIN property_type pt ON ws.property_type = pt.property_type_id
LEFT JOIN location loc ON ws.location = loc.location_id
WHERE kitchen = 'Y'

1 rows affected.


[(244,)]